packages/shared/lib/services/environment.service.ts

```diff
@@ -479,6 +495,19 @@ class EnvironmentService {
 
         return globalCallbackUrl;
     }
+
+    async deleteEnvironment(id: number): Promise<Result<void, EnvironmentError>> {
+        try {
+            const result = await db.knex.from<DBEnvironment>(TABLE).where({ id }).del();
+            if (result !== 0) {
+                return Ok(undefined);
+            }
+        } catch {
+            // Empty catch block
+        }
+        // Error if no environment was found OR exception was thrown
+        return Err(new EnvironmentError({ code: 'deletion_failed', message: 'Failed to delete environment' }));
+    }
```

Comment:
We actually want to differentiate the two for error reporting. 

```suggestion
        try {
            const result = await db.knex.from<DBEnvironment>(TABLE).where({ id }).del();
            if (result !== 0) {
                return Ok(undefined);
            }
             // Error if no environment was found OR exception was thrown
             return Err(new Error('blabla'));
        } catch (err) {
            return Err(new Error('blabla', { cause: err }), 
        }
       
    }
```

Also we don't want to introduce `EnvironmentError` rn. I appreciate the initiative, and it's probably going to look similar to that at some point we are actually trying to have global revamp of the error and it's a bit too soon. Using generic Error should be fine